lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -391,16 +418,15 @@ private static Domain getDomain(Type type, DictionaryDescriptor dictionaryDescri
             return Domain.all(type);
         }
 
-        Domain domain = Domain.none(type);
         int dictionarySize = dictionaryPage.get().getDictionarySize();
         DictionaryValueConverter converter = new DictionaryValueConverter(dictionary);
         Function<Integer, Object> convertFunction = converter.getConverter(columnDescriptor.getPrimitiveType());
+        List<Object> values = new ArrayList<>();
         for (int i = 0; i < dictionarySize; i++) {
-            Object value = convertFunction.apply(i);
-            domain = domain.union(getDomain(type, value, value, true, timeZone));
+            values.add(convertFunction.apply(i));
         }
 
-        return domain;
+        return getDomain(type, values, values, true, timeZone);
```

Comment:
there is dedicated code path in `io.trino.spi.predicate.SortedRangeSet#of` for handling singleton ranges like this, which should result in faster construction time and more efficient representation later.

Reply:
I know, but I can't currently think of a way to take advantage of it without duplicating the `getDomain` method and specialize it to singleton ranges.

Reply:
We can have `ValueSet.ofRanges` realize all ranges are singletons and take different path.
Or, if we don't want such change for everyone, we could inject ValueSet constructor that would do that.

I am fine with current code though, let's have a TODO comment about known room for improvement (and perhaps an issue)

Reply:
I'll add a TODO for now so we can get this code in. I'm playing with a couple of other alternatives for improving that case in parallel that I may put up later once it's in good shape.